library/events/json_message_event.nim

```diff
@@ -0,0 +1,42 @@
+
+import
+    std/json
+import
+    ../../waku/v2/waku_core/message/message,
+    json_signal_event
+
+type JsonMessage = ref object
+  # https://rfc.vac.dev/spec/36/#jsonmessage-type
+  payload: string
+  contentTopic: string
+  version: uint
+  timestamp: int64
+
+type JsonMessageEvent* = ref object of JsonSignal
+    pubsubTopic*: string
+    messageId*: string
+    wakuMessage*: JsonMessage
+
+proc new*(T: type JsonMessageEvent,
+          pubSubTopic: string,
+          msg: WakuMessage): T =
+  # Returns a WakuMessage event as indicated in
+  # https://rfc.vac.dev/spec/36/#jsonmessageevent-type
+
+  var payload = newString(len(msg.payload))
+  copyMem(addr payload[0], unsafeAddr msg.payload[0], len(msg.payload))
```

Comment:
I'm curious about this line. Why is it necessary to copy the payload like this instead of doing it directly?

Reply:
Good point! I will try to directly use `seq[byte]` instead

Reply:
I think I will leave it as is because if I cahnge it to use `seq[byte]` we don't get a readable base64 string. In this case, it appears sth like `"payload":[97,71,57,115,89,81,61,61]`